pkg/manager/router/router_score.go

```diff
@@ -318,6 +375,32 @@ func (router *ScoreBasedRouter) OnBackendChanged(backends map[string]*BackendHea
 	if len(backends) > 0 {
 		router.updateServerVersion()
 	}
+	if router.enablePause && len(backends) > 0 {
+		pause := true
+		for be := router.backends.Front(); be != nil; be = be.Next() {
+			backend := be.Value
+			if backend.Healthy() {
+				pause = false
+				break
+			}
+		}
+		if pause {
+			router.logger.Info("last backend become unhealthy, notify connections to pause")
+			for be := router.backends.Front(); be != nil; be = be.Next() {
+				backend := be.Value
+				backend.connScore--
+				for ele := backend.connList.Front(); ele != nil; ele = ele.Next() {
+					conn := ele.Value
+					switch conn.phase {
+					case phasePauseNotify, phaseRedirectNotify:
+						continue
```

Comment:
If a connection is redirecting and doesn't pause, you need to handle it. E.g. notify it later.

Reply:
In serverless scenario. If conn is redirecting while pausing, it indicates the backend become alive while pausing. I think it's OK to just let it migrate to the new backend. Can I just leave a log here? And maybe we can handle it better   when pause be triggered by other events in the future.

Reply:
One question, if a connection is in a txn but it has not executed any SQL for 5 min (that is, it never executes `COMMIT`), will the backend be recycled? 
If it will be, then what if it continues the transaction after the backend is recycled?

Reply:
Nope, If a connection is in a transaction the backend will not quit until the transaction be commit. Controlled by inTransCount in this line. https://github.com/tidbcloud/tidb-cse/blob/release-7.1-keyspace/standby/idle_watcher.go#L83

Reply:
OK, then it'll be fine. Leave a comment here to clarify that when a connection is redirecting, the backend won't be paused.